### PR TITLE
Method to create customer

### PR DESCRIPTION
**Pull request by carloslopdev**
This PR adds a method to create a customer, following this Modulr API endpoint reference:

https://modulr.readme.io/reference/createcustomer

### CODE REVIEW

File: lib/modulr/api/customers_service.rb [hunk 1]
```diff
@@ -1,12 +1,40 @@
...
+        payload[:expectedMonthlySpend] = opts[:expected_monthly_spend] if opts[:expected_monthly_spend]
+        payload[:associates] = opts[:associates] if opts[:associates]
+        payload[:documentInfo] = opts[:document_info] if opts[:document_info]
+        payload[:provisionalCustomerId] = opts[:provisional_customer_id] if opts[:provisional_customer_id]
+        payload[:customerTrust] = opts[:customer_trust] if opts[:customer_trust]
+        payload[:taxProfile] = opts[:tax_profile] if opts[:tax_profile]
+
+        response = client.post("/customers", payload)
+        Resources::Customers::Customer.new(response.env[:raw_body], response.body)
+      end
```
Comment (Nachooc5) [inline lib/modulr/api/customers_service.rb:35]:
```suggestion
      def create(**opts)
        response = client.post("/customers", opts)
        Resources::Customers::Customer.new(response.env[:raw_body], response.body)
      end
```

Comment (carloslopdev) [reply lib/modulr/api/customers_service.rb:35]: I totally agree with that, but the thing is, in `lib/modulr/api/accounts_service.rb` (code below) we are already doing that kind of renaming, so for consistency I followed the same pattern...

```rb
def create(customer_id:, currency:, product_code:, **opts)
    payload = {
        currency: currency,
        productCode: product_code,
    }

    payload[:externalReference] = opts[:external_reference] if opts[:external_reference]

    response = client.post("/customers/#{customer_id}/accounts", payload)
    Resources::Accounts::Account.new(response.env[:raw_body], response.body)
end
```

**Review by Nachooc5 — APPROVED** [review]

File: lib/modulr/api/customers_service.rb [hunk 2]
```diff
@@ -1,12 +1,40 @@
...
       end
+
+      def create(type:, legal_entity:, **opts)
+        payload = {
+          type: type,
+          legalEntity: legal_entity,
+        }
+
+        payload[:externalReference] = opts[:external_reference] if opts[:external_reference]
+        payload[:name] = opts[:name] if opts[:name]
```
Comment (Franx0) [inline lib/modulr/api/customers_service.rb:20]: Some fields are mandatory or conditioanl mandatory. Name is one of them so we should control it from client? 🤔

Comment (carloslopdev) [reply lib/modulr/api/customers_service.rb:20]: Yep, but after discussing, we are more on the side of reducing the amount of logic in this gem, the way that @Nachooc5 comments here: https://github.com/devengoapp/modulr-ruby/pull/15#discussion_r1162480348.

So introducing that logic would be a step in the opposite direction, would make the gem more dependant of Modulr.

Of course, open to discussion as usual.

**Pull request merged**